esrally/driver/runner.py

```diff
@@ -2683,6 +2684,49 @@ def __repr__(self, *args, **kwargs):
         return "sql"
 
 
+class Downsampling(Runner):
+    """
+    Executes a downsampling operation creating the target index and aggregating data in the source index on the @timestamp field.
+    """
+
+    async def __call__(self, es, params):
+
+        request_params, request_headers = self._transport_request_params(params)
+        body = mandatory(params, "body", self)
+
+        fixed_interval = body.get("fixed-interval")
+        if fixed_interval is None:
+            raise exceptions.DataError(
+                "Parameter source for operation 'downsampling' did not provide the mandatory parameter 'body.fixed-interval'. "
+                "Add it to your parameter source and try again."
+            )
+
+        source_index = body.get("source-index")
+        if source_index is None:
+            raise exceptions.DataError(
+                "Parameter source for operation 'downsampling' did not provide the mandatory parameter 'body.source-index'. "
+                "Add it to your parameter source and try again."
+            )
+
+        target_index = body.get("target-index")
+        if target_index is None:
+            raise exceptions.DataError(
+                "Parameter source for operation 'downsampling' did not provide the mandatory parameter 'body.target-index'. "
+                "Add it to your parameter source and try again."
+            )
+
+        path = "/" + source_index + "/_rollup/" + target_index
```

Comment:
I will need to change the endpoint to include `_downsample` instead of `_rollup` as soon as PR https://github.com/elastic/elasticsearch/pull/89809 is merged.

Reply:
FYI we tend to prefer f-strings for this type of code
```suggestion
        path = f"/{source_index}/_rollup/{target_index}"
```